### PR TITLE
Allow tag to limit pages scraped

### PR DESCRIPTION
**Pull request by Gathondu**
#### What does this PR do?
Allow for using `small_batch` tag for development to only scrape limited no of pages
Allow to save to local file system or S3 bucket
Define S3 to bucket
Update README to reflect the changes
#### Description of Task to be completed?
Add "flag" functionality. Suggested flag: `small_batch`.
Allow defining S3 bucket in config.
Allow local storage of files in development mode.
#### How should this be manually tested?
run 'nosests --nocapture`

### CODE REVIEW

File: healthtools/config.py [hunk 1]
```diff
@@ -12,11 +12,12 @@
...
     "aws_secret_access_key": os.getenv("MORPH_AWS_SECRET_KEY"),
     "region_name": os.getenv("MORPH_AWS_REGION", "eu-west-1"),
-    "s3_bucket": os.getenv("S3_BUCKET", "cfa-healthtools-ke")
+    "s3_bucket": os.getenv("S3_BUCKET", None)
 }
 
 ES = {
-    "host": os.getenv("ES_HOST", None),
+    "host": os.getenv("ES_HOST", '127.0.0.1'),
+    "port": os.getenv("ES_PORT", '9200'),
```
Comment (DavidLemayian) [inline healthtools/config.py]: @Gathondu Single vs double quote..

File: healthtools/config.py [hunk 2]
```diff
@@ -26,6 +27,7 @@
 
 TEST_DIR = os.getcwd() + "/healthtools/tests"
 
-SLACK = {
-    "url": os.getenv("WEBHOOK_URL")
-    }
+BATCH = 5
+HF_BATCH = 100  # batch for health facilities
+
+DATA_DIR = os.getcwd() + "/healthtools/"
```
Comment (DavidLemayian) [inline healthtools/config.py]: Is `/data` or `/healthtools`? - https://github.com/CodeForAfrica-SCRAPERS/healthtools_ke/pull/13/files#diff-04c6e90faac2675aa89e2176d2eec7d8R30

File: healthtools/scrapers/base_scraper.py [hunk 3]
```diff
@@ -26,21 +28,31 @@ def __init__(self):
...
-                verify_certs=True,
-                connection_class=RequestsHttpConnection,
-                serializer=JSONSerializerPython2()
-            )
-        else:
-            self.es_client = Elasticsearch('127.0.0.1')
+        self.small_batch = True if 'small_batch' in sys.argv else False
+        try:
+            # client host for aws elastic search service
+            if 'aws' in ES['host']:
```
Comment (DavidLemayian) [inline healthtools/scrapers/base_scraper.py]: Single vs double.

File: healthtools/scrapers/base_scraper.py [hunk 4]
```diff
@@ -26,21 +28,31 @@ def __init__(self):
...
+                # set up authentication credentials
+                awsauth = AWS4Auth(AWS["aws_access_key_id"], AWS["aws_secret_access_key"], AWS["region_name"], 'es')
```
Comment (DavidLemayian) [inline healthtools/scrapers/base_scraper.py]: Single vs double quote

File: healthtools/scrapers/base_scraper.py [hunk 5]
```diff
@@ -26,21 +28,31 @@ def __init__(self):
...
+                    hosts=ES['host'],
+                    port=ES['port'],
+                    http_auth=awsauth,
+                    use_ssl=True,
+                    verify_certs=True,
+                    connection_class=RequestsHttpConnection,
+                    serializer=JSONSerializerPython2()
+                )
+            else:
+                self.es_client = Elasticsearch("{}:{}".format(ES['host'], ES['port']))
```
Comment (DavidLemayian) [inline healthtools/scrapers/base_scraper.py]: Single vs double quote.

File: healthtools/config.py (hunk 2 above)
Comment (DavidLemayian) [reply healthtools/config.py]: I notice now that you're saving it in the package.. (`healthtools/data`) I would advise against that and instead go for `data` only and have that as the `DATA_DIR`.

File: healthtools/config.py [hunk 6]
```diff
@@ -26,6 +27,7 @@
 
 TEST_DIR = os.getcwd() + "/healthtools/tests"
 
-SLACK = {
-    "url": os.getenv("WEBHOOK_URL")
-    }
+BATCH = 5
+HF_BATCH = 100  # batch for health facilities
```
Comment (DavidLemayian) [inline healthtools/config.py]: Let's call these `SMALL_BATCH` and `SMALL_BATCH_HF`, and add inline comments.

File: healthtools/scrapers/base_scraper.py [hunk 7]
```diff
@@ -73,7 +84,7 @@ def scrape_site(self):
                 self.print_error("ERROR: scrape_site() - source: {} - page: {} - {}".format(url, page_num, err))
                 continue
         print "[{0}] - Scraper completed. {1} documents retrieved.".format(
-            datetime.now().strftime('%Y-%m-%d %H:%M:%S'), len(all_results))
+            datetime.now().strftime('%Y-%m-%d %H:%M:%S'), len(all_results)/2)  # don't count indexing data
```
Comment (DavidLemayian) [inline healthtools/scrapers/base_scraper.py]: Good catch.

**Review by DavidLemayian — CHANGES_REQUESTED** [review]

File: healthtools/config.py (hunk 1 above)
Comment (Gathondu) [reply healthtools/config.py]: Arrrgh 🤦‍♂️ . I mentally settled for single quotes, but in this project most of the quotes were double

File: healthtools/config.py (hunk 6 above)
Comment (Gathondu) [reply healthtools/config.py]: 👍

File: healthtools/config.py (hunk 2 above)
Comment (Gathondu) [reply healthtools/config.py]: 👍

**Comment by Gathondu** [comment]
@DavidLemayian changes have been made 😄

File: healthtools/scrapers/base_scraper.py [hunk 8]
```diff
@@ -26,35 +28,45 @@ def __init__(self):
...
+            else:
+                self.es_client = Elasticsearch("{}:{}".format(ES["host"], ES["port"]))
+        except Exception as err:
+            self.print_error("[{}] - ERROR: Invalid Parameters For ES Client".format(
+                datetime.now().strftime("%Y-%m-%d %H:%M:%S")))
+        # if to save locally create relevant directories
+        if not AWS["s3_bucket"] and not os.path.exists(DATA_DIR):
+            os.mkdir(DATA_DIR)
+            os.mkdir(DATA_DIR + "data")
+            os.mkdir(DATA_DIR + "data/archive")
```
Comment (DavidLemayian) [inline healthtools/scrapers/base_scraper.py]: Considering already `/data`, do these become redundant?

**Review by DavidLemayian — CHANGES_REQUESTED** [review]

**Comment by DavidLemayian** [comment]
@Gathondu one more. ☝️

File: healthtools/scrapers/base_scraper.py (hunk 8 above)
Comment (Gathondu) [reply healthtools/scrapers/base_scraper.py]: @DavidLemayian I tried not to change the s3 keys, but I've got an idea of how to work around that

**Comment by Gathondu** [comment]
@DavidLemayian redundant files should be history now.

**Pull request merged**